src/test/scala/za/co/absa/commons/reflect/ReflectionUtilsSpec.scala

```diff
@@ -136,6 +136,14 @@ class ReflectionUtilsSpec extends AnyFlatSpec with Matchers with MockitoSugar {
     ReflectionUtils.extractFieldValue[Boolean](Foo, "x") should be(Foo.x)
   }
 
+  // this class must be outside the test to trigger the exception
+  case class WrappingClass(x: ClassWithCyclicAnnotation)
+
+  it should "fallback to Java reflection when Scala one fails via return type" in {
+    val obj = WrappingClass(new ClassWithCyclicAnnotation)
+    ReflectionUtils.extractFieldValue[ClassWithCyclicAnnotation](obj, "x") should not be null
```

Comment:
just to shut Codacy up...
```suggestion
    val obj = WrappingClass(new ClassWithCyclicAnnotation)
    // scalastyle:off null
    ReflectionUtils.extractFieldValue[ClassWithCyclicAnnotation](obj, "x") should not be null
    // scalastyle:on null
```